### PR TITLE
Defined REST APIs to update PromotionResult for existing deployments

### PR DESCRIPTION
**Pull request by maheshkelkar**
References #19

### CODE REVIEW

**Comment by maheshkelkar** [comment]
@tommysullivan this is relevant to our discussion today

File: features/api/promotionResult/updating.feature [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  As a RESTful client or user
+  I should be able to update promotionResult for the existing deployments in the system
+
+
+  @wip
+  Scenario: Updating the PromotionResult for a deployment
+
+    Given there is a deployment
+    When I PUT "/api/deployments/1/promotions" with:
```
Comment (rtyler) [inline features/api/promotionResult/updating.feature]: We're using `POST` everywhere for adding to a collection, see the artifact create APIs

File: features/api/promotionResult/updating.feature [hunk 2]
```diff
@@ -0,0 +1,48 @@
+Feature: PromotionResult UPDATE APIs
+
+  As a RESTful client or user
+  I should be able to update promotionResult for the existing deployments in the system
+
+
+  @wip
+  Scenario: Updating the PromotionResult for a deployment
```
Comment (rtyler) [inline features/api/promotionResult/updating.feature]: This is *adding* a promotion result

File: features/api/promotionResult/updating.feature [hunk 3]
```diff
@@ -0,0 +1,48 @@
+Feature: PromotionResult UPDATE APIs
```
Comment (rtyler) [inline features/api/promotionResult/updating.feature]: There are no updates of promotion results here, we're creating new promotion result entries

File: features/api/promotionResult/updating.feature [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        "name"  : "jenkins-smoke",
+        "status" : "SUCCESS",
+        "resultsUrl" : "http://local.lookout.com/jenkins/job-id/2/results"
+      }
+    """
+    Then the response should be 404
+
+
+  @wip @error
+  Scenario: Updating the PromotionResult for promotion that doesn't exist in deployment
```
Comment (rtyler) [inline features/api/promotionResult/updating.feature]: This is confusing with how it's worded, I suggest changing to sometihng like:

```cucumber
Scenario:  Adding a result for a promotion which is not assocaited with the deployment
```

**Comment by rtyler** [comment]
This looks great to me, I'm going to mergey merge

**Pull request merged**